### PR TITLE
Michael's Instagram Challenge

### PR DESCRIPTION
**Pull request by mcsuGH**
Implemented most of the functionality (can make post, comment on posts, like on posts). Users can sign in and sign out and conditionals are used to show content depending on whether or not the user is signed in or not (for example, if you are signed in, the Sign In button at the navigation bar becomes a Log Out, or if you are already logged in and you go to the Login Page it will tell you you are already logged in.)

Will need to add functionality to upload images to database (currently only allows URL of external image hosts like imgur) and still need to add more CSS to make it look nicer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,41 +1,68 @@
...
+used the above to upload and retrieve images in mongoDB - abandoned for now, would make my req.body empty for some reason making my posts empty.
 
-You can also lint Javascript, CSS, and ERB — feel free to research this. These
-will help you to train yourself to produce cleaner code — and will often alert
-you to mistakes or mishaps!
+## Functionality yet to be added
+- add more CSS - currently only added a bar at the top and a background for the post images, displaying the images in rows of 3.
+- make it so each user can only like a post once (would just need to create a new collection for likes inside database with foreign keys for the post and the user who liked the post)
+- add filter function
+- add the ability to upload images onto the database rather than using a external image host
```
Comment (JordanManu) [inline README.md:68]: Great readme instructions, very easy to follow

File: controllers/posts.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  },
+  Delete: (req, res) => {
+    Post.findByIdAndRemove(req.params.id, 
+      function(err, docs) {
+        if(err) res.json(err);
+        else res.status(201).redirect("/posts");
+      })
+  }
+}  
+
```
Comment (JordanManu) [inline controllers/posts.js:74]: all v nice!

File: cypress/integration/post_image_spec.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    cy.get("#password").type("password");
+    cy.get("#submit").click();
+    cy.get(".newpost").click();
+
+    cy.get("#url").type("https://i.imgur.com/PpUyft2.jpeg");
+    cy.get("#caption").type("this is a picture");
+    cy.get(".createpost").click();
+
+    cy.url().should("include", "/posts");
+    cy.get('div[class="posts"]').find(".postimage").should('be.visible');
```
Comment (JordanManu) [inline cypress/integration/post_image_spec.js:14]: I like this test!

File: cypress/integration/sign_out_spec.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    // sign up
+    cy.visit("/users/new");
+    cy.get("#email").type("someone3@example.com");
+    cy.get("#password").type("password");
+    cy.get("#submit").click();
+    cy.get(".logout").click();
+
+    cy.url().should("include", "/sessions");
+    cy.should("not.contain", "Hey someone3@example.com");
+  });
```
Comment (JordanManu) [inline cypress/integration/sign_out_spec.js:12]: Good test

File: cypress/integration/sign_up_spec.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  });
+
+  it("A user cannot sign up with an email that is already in use", () => {
+    // credentials already used in previous test
+    cy.visit("/users/new");
+    cy.get("#email").type("someone2@example.com");
+    cy.get("#password").type("password");
+    cy.get("#submit").click();
+
+    cy.contains("Email already in use");
```
Comment (JordanManu) [inline cypress/integration/sign_up_spec.js:19]: nice!

File: spec/models/post.test.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    var post = new Post({ user: "627b88582de61f0e5db7f4ca", url: "http://someimg.com" })
+
+    post.save((err) => {
+      expect(err).toBeNull();
+      
+      Post.find((err, posts) => {
+        expect(err).toBeNull();
+
+        expect(posts[0].createdAt).toBeDefined();
+        done();
```
Comment (JordanManu) [inline spec/models/post.test.js:62]: Great tests!

File: views/layout.hbs [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+          <button class = "logout">Log Out</button>
+        </form>
+      {{else}}
+        <form action='/sessions/new'>
+          <button class = "loginpage">Sign In</button>
+        </form>
+      {{/if}}
+    </div>
+    {{#if user}}
+      Hey {{user.email}}
```
Comment (JordanManu) [inline views/layout.hbs:16]: Really like this feature

File: views/posts/id.hbs [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+<br>
+Write comment: 
+<div>
+  <form id="comment-form" action='/posts/{{post.id}}' method="POST">
+    <div>
+      <label for="comment">Comment</label>
+      <input type="text" id="comment" name="comment" placeholder="write comment" required>
+    </div>
+    <div>
+      <button class ='createcomment'>Comment</button>
```
Comment (JordanManu) [inline views/posts/id.hbs:38]: Real good

File: views/posts/id.hbs [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    </div>
+  </form>
+</div>
+<br>
+<form action = '/posts'>
+  <button class = "back">back</button>
```
Comment (JordanManu) [inline views/posts/id.hbs:44]: I like the back button

File: views/sessions/new.hbs [hunk 10]
```diff
@@ -0,0 +1,17 @@
+{{#if user}}
+You are already logged in.
```
Comment (JordanManu) [inline views/sessions/new.hbs:2]: Great

**Review by JordanManu — COMMENTED** [review]
Great project! It works really well, I love the comment feature that displays who commented and the time it was posted. Edge cases were covered and great test coverage!
